sympy/polys/rings.py

```diff
@@ -2531,15 +2531,145 @@ def compose(f, x, a=None):
 
         return poly
 
+
+    def coeff_wrt(self, x, dg):
+        """
+        Coefficient of ``f`` with respect to ``x**deg``.
+
+        Treating ``f`` as a univariate polynomial in ``x`` this finds the
+        coefficient of ``x**deg`` as a polynomial in the other generators.
+
+        Parameters
+        ==========
+
+        p : sympy.Poly
+            The polynomial to compute the sparse coefficient for.
+        x : int
+            The symbol to compute the coefficient for.
+        dg : int
+            The degree of the monomial to compute the coefficient for.
+
+        Returns
+        =======
+
+        sympy.Expr
+            The expression of the polynomial ``f`` at the given symbol and degree.
+
+        Examples
+        ========
+
+        >>> from sympy.polys import ring, ZZ
+        >>> _, x, y, z = ring("x, y, z", ZZ)
+        >>> p = 2*x**4 + 3*y**4 + 10*z**2 + 10*x*z**2
+        >>> dg = 2
+        >>> x = 2
+        >>> p.coeff_wrt(x, dg)
+        10*x + 10
```

Comment:
Rather than using `x=2` I would just make two examples like:
```
        Find the coefficient of `z^2` as a polynomial in `x` and `y`:

        >>> p.coeff_wrt(z, 2)
        10*x + 10
        >>> p.coeff_wrt(2, 2)  # Using the generator index
        10*x + 10
```